server.js

```diff
@@ -2,37 +2,82 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import netflixData from './data/netflix-titles.json'
+import NetflixShow from './models/show'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+if (process.env.RESET_DB) {
+  console.log('Resetting Database!')
+  const seedDatabase = async () => {
+    await NetflixShow.deleteMany({})
+    netflixData.forEach((show) => {
+      new NetflixShow(show).save()
+    })
+  }
+  seedDatabase()
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable'})
+  }
+})
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(`<p>Netflix show data with endpoints: <br> /netflixshows<br> /netflixshows?query="Your Query"<br> /netflixshows/type?type="Movie/TV Show"<br> /netflixshows/rating?rating="TV-Y/TV-Y7/TV-Y7-FV/TV-G/G/TV-PG/PG/PG-13/TV-14/TV-MA/R"<br> /netflixshows/title/"Your Title"<p>`)
+})
+
+app.get('/netflixshows', async (req, res) => {
+  const { query } = req.query
+  const queryRegEx = new RegExp(query, 'i') 
+  const netflixShows = await NetflixShow.find({ title: queryRegEx }).sort({title: 1})
+  res.json(netflixShows)
+})
+ 
+app.get('/netflixshows/type', async (req, res) => {
+  const { type } = req.query
+  let shows = await NetflixShow.find()
+  if (type) { 
+    shows = shows.filter((show) => show.type.toString() === type)
+  }
+  if (shows.length === 0) {
+    res.status(404).send('Please select Movie or TV Show to filter on "type"')
+  }
+  res.json(shows)
+})
```

Comment:
It would be more RESTful if this query was in the `/netflixshows` endpoint. And when working with mongo, try to stick to mongo methods like `find`, instead of regular javascript methods like `filter()`. (And maybe add regex for case insensitive here to make it possible to write in lower case)